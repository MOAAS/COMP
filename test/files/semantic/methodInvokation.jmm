import static io.println(int) void;
import static io.println(int) boolean;

import Quicksort;
import static Quicksort.do;

class Simple extends Quicksort {
    int simpleVar;
    int verySimpleVar;


    public static void main(String[] arg){
        int a;
        int b;
        Simple simple;
        Simple io;
        int c;
        a = 30;
        b = 0-10;
        simple = new Simple();
        io = new Simple();

        simple.add(c, d);

        a = 1 + simple.getBool();
        a = verySimpleVar;
        verySimpleVar = 3;

        a.add(1,2);

        c = person.getCar(simple.func0()).getLicense(noSymbol1.func1().func2(noSymbol2), simple.func3(), new Simple().length).getNumber(simple.func4(a, b));

        simpleVar = true;

        io.println(c);
        Simple.add(1, 2, 3);
        Simple.add(1, 2);
        io.simple1(Simple);
        io.quicksort1(Quicksort);
    }

    public int add(int a, int b){
        this.do();
        verySimpleVar = 3;
        return a + b;
    }

    public Simple simple1(Simple s) {
        return Simple;
    }

    public Quicksort quicksort1(Quicksort s) {
        return Quicksort;
    }

    public int println() {
        return true;
    }

    public boolean getBool() {
        return true;
    }
}