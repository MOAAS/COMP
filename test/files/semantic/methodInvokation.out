Warning: Duplicate import declaration "io.println(int)" at line 2, column 1.
Error: Variable "c" has not been initialized at line 23, column 20.
Error: Symbol "d" does not exist at line 23, column 23.
Error: Right operand of + is not an int. Found: boolean at line 25, column 17.
Error: Symbol "verySimpleVar" does not exist at line 26, column 13.
Error: Symbol "verySimpleVar" does not exist at line 27, column 9.
Error: Class "int" does not contain method "add(int,int)" at line 29, column 9.
Error: Symbol "person" does not exist at line 31, column 13.
Error: Class "Simple" does not contain method "func0()" at line 31, column 27.
Error: Symbol "noSymbol1" does not exist at line 31, column 54.
Error: Symbol "noSymbol2" does not exist at line 31, column 78.
Error: Class "Simple" does not contain method "func3()" at line 31, column 90.
Error: Tried to call ".length" on non array. Found type: Simple at line 31, column 110.
Error: Class "Simple" does not contain method "func4(int,int)" at line 31, column 137.
Error: Symbol "simpleVar" does not exist at line 33, column 9.
Error: Variable "c" has not been initialized at line 35, column 20.
Error: Class "Simple" does not contain static method "add(int,int,int)" at line 36, column 9.
Error: Method "Simple.add(int,int)" is not static at line 37, column 9.
Error: Simple is not valid as a method argument (of function "simple1") at line 38, column 20.
Error: Quicksort is not valid as a method argument (of function "quicksort1") at line 39, column 23.
Error: Method "Simple.do()" is static at line 43, column 9.
Error: Invalid return expression. Required: Simple, Provided: class at line 49, column 16.
Error: Invalid return expression. Required: Quicksort, Provided: class at line 53, column 16.
Error: Invalid return expression. Required: int, Provided: boolean at line 57, column 16.
